src/components/KTreeList/KTreeItem.vue

```diff
@@ -70,7 +71,23 @@ const slots = useSlots()
 
 const hasIcon = computed((): boolean => !props.hideIcons || !!slots['item-icon'])
 
-const handleClick = () => {
+const handleClick = (event: any) => {
+  if (event.target) {
+    const ignoredElements = ['a', 'button:not([data-tree-item-trigger])', 'label', 'input', 'select']
+
+    // check whether target is an ignored elem
+    if (ignoredElements.includes(event.target.tagName.toLowerCase())) {
+      return
+    }
```

Comment:
Can we make `ignoredElements` a selector like `a, button:not([data-tree-item-trigger]), label, input, select` use `event.target.closest(ignored) !== null` here?

As `closest` starts searching from the element itself, we don't have to match the element itself and parents separately.

Ps. we can add `textarea` to the list, and maybe add `:not(:disabled)` to form elements to make it more complete.

Reply:
Removed the unneeded element check and added the `:not(:disabled)` selector.

As for `textarea` selector, we don't need it in this list as no one should try to slot in textarea into the tree list item label - the component is way too small for that.